clang/lib/CConv/ArrayBoundsInferenceConsumer.cpp

```diff
@@ -166,7 +166,7 @@ static bool isAllocatorCall(Expr *E) {
 }
 
 static bool isStringLiteral(Expr *E) {
-  return dyn_cast<StringLiteral>(removeAuxillaryCasts(E));
+  return isa<StringLiteral>(removeAuxillaryCasts(E));
```

Comment:
If `E` or `removeAuxillaryCasts(E)` is null then `isa` would result in an assertion. It's better to either assert that E is non-null or check it explicitly for non-null.